cdap-ui/app/cdap/components/AbstractWidget/MultiStageSelector/index.tsx

```diff
@@ -58,28 +60,44 @@ const MultiStageSelectorBase: React.FC<IMultiStageSelectorProps> = ({
 }) => {
   const inputSchema = objectQuery(extraConfig, 'inputSchema');
   const inputStages = getInputStages(inputSchema);
-  const { delimiter } = widgetProps;
-  const multiSelectWidgetProps = {
-    delimiter,
-    options: inputStages,
-    showSelectionCount: false,
-    emptyPlaceholder: 'Select input stage',
-  };
+  const { delimiter, singleSelectOnly } = widgetProps;
+
   if (!inputStages.length) {
     return (
       <div className={classes.emptyMessageContainer}>
         <div className={classes.emptyMessage}>No input stages</div>
       </div>
     );
   }
-  return (
-    <MultiSelect
-      value={value}
-      onChange={onChange}
-      widgetProps={multiSelectWidgetProps}
-      disabled={disabled}
-    />
-  );
+
+  if (singleSelectOnly || false) {
+    const selectWidgetProps = {
+      options: inputStages.map((stage) => stage.label),
```

Comment:
Can we reuse the same `getInputStages` function here?

Reply:
not quite...Multselect expects options to be `{'id': '.', 'value': '.'}` meanwhile Select expects  `{'label': '.', 'value': '.'}` so I would need to move this logic into that function anyway to change the format

Reply:
Haa! That's a bad API. Ok can you file a jira for this. It's ok to merge now but that's bad.